airflow-core/src/airflow/models/taskinstance.py

```diff
@@ -326,6 +327,19 @@ def clear_task_instances(
             if dr.state in State.finished_dr_states:
                 dr.state = dag_run_state
                 dr.start_date = timezone.utcnow()
+                if TYPE_CHECKING:
+                    assert dag  # todo: change signature so this is required
```

Comment:
Since this function has a DagBag object, I’d just auto-populate the variable instead. This is actually done above on line 288 for another situation.

Reply:
i changed the signature.

there's not much point in passing dag in as a param to this function since the TIs can be from different dags.

moreover, they can be from different serdags associated with the same dag -- and that in particular is what makes this complicated and required me to go and use the scheduler dag bag object